File: UniBlock/src/main/scala/helper/crypto/RandomHelper.scala
package it.unifi.nave.uniblock
package helper.crypto

import java.security.SecureRandom

object RandomHelper {
  private val SECURE_RANDOM = new SecureRandom

  def generateRandom(length: Int): Array[Byte] = {
    val random = new Array[Byte](length)
    SECURE_RANDOM.nextBytes(random)
    random
  }

}

File: src/main/java/it/unifi/nave/uniblock/helper/crypto/RandomHelper.java
package it.unifi.nave.uniblock.helper.crypto;

import java.security.SecureRandom;

public class RandomHelper {
    private static final SecureRandom SECURE_RANDOM = new SecureRandom();

    public static byte[] generateRandom(int length) {
        var random = new byte[length];
        SECURE_RANDOM.nextBytes(random);
        return random;
    }
}
